org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetEditingWizardPreviewPage.java

```diff
@@ -65,13 +75,37 @@ public void createControl(Composite parent) {
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
 
-		// TODO: create preview page control here
 		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create preview page control here");
+
+		createPreviewViewer(container);
+
+		populateUi();
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
 	}
+
+	private void createPreviewViewer(Composite container) {
+		Composite parent = new Composite(container, SWT.NONE);
+		parent.setLayout(new FillLayout());
+
+		VerticalRuler ruler = new VerticalRuler(0);
+		SourceViewer editor = new SourceViewer(parent, ruler, SWT.NONE);
+		editor.configure(new XmlConfiguration(new ColorManager()));
+
+		document = new Document();
+		IDocumentPartitioner partitioner = new FastPartitioner(new XmlPartitionScanner(),
+				new String[] {XmlPartitionScanner.XML_TAG, XmlPartitionScanner.XML_COMMENT});
+		partitioner.connect(document);
+		document.setDocumentPartitioner(partitioner);
+		editor.setDocument(document);
+
+		editor.setEditable(false);
+	}
+
+	private void populateUi() {
+		document.set("<jfragent>\n</jfragent>"); // TODO: generate real output preview from this.preset
```

Comment:
As said in TODO output preview is not generated

Reply:
Note more of a note: this requires XML serialization to be implemented first.